diff --git a/cmake/public/LoadHIP.cmake b/cmake/public/LoadHIP.cmake
index cae0ca62f23..ed2bd0862f6 100644
--- a/cmake/public/LoadHIP.cmake
+++ b/cmake/public/LoadHIP.cmake
@@ -93,24 +93,25 @@ if(HIP_FOUND)
   # hip (lower-case) package. Both are probed above and will be in
   # ROCM_INCLUDE_DIRS if available.
   find_file(ROCM_VERSION_HEADER_PATH
-    NAMES rocm-core/rocm_version.h
+    NAMES
+      rocm-core/rocm_version.h  # Higher priority
+      hip/hip_version.h         # Lower priority
     NO_DEFAULT_PATH
     PATHS ${ROCM_INCLUDE_DIRS}
   )
-  set(ROCM_LIB_NAME "ROCM")
-  if(NOT ROCM_VERSION_HEADER_PATH)
-    find_file(ROCM_VERSION_HEADER_PATH
-      NAMES hip/hip_version.h
-      NO_DEFAULT_PATH
-      PATHS ${ROCM_INCLUDE_DIRS}
-    )
-    set(ROCM_LIB_NAME "HIP")
-  endif()
+
   if(NOT ROCM_VERSION_HEADER_PATH)
     message(FATAL_ERROR "Could not find hip/hip_version.h or rocm-core/rocm_version.h in ${ROCM_INCLUDE_DIRS}")
   endif()
+
   get_filename_component(ROCM_HEADER_NAME ${ROCM_VERSION_HEADER_PATH} NAME)
 
+  if(ROCM_HEADER_NAME STREQUAL "rocm_version.h")
+    set(ROCM_LIB_NAME "ROCM")
+  else()
+    set(ROCM_LIB_NAME "HIP")
+  endif()
+
   if(EXISTS ${ROCM_VERSION_HEADER_PATH})
     set(ROCM_HEADER_FILE ${ROCM_VERSION_HEADER_PATH})
   else()
